libraries/lib-file-formats/AcidizerTags.h

```diff
@@ -10,11 +10,45 @@
 **********************************************************************/
 #pragma once
 
+#include <cassert>
+#include <optional>
+
 namespace LibFileFormats
 {
+/*!
+ * @brief Models how SoundForge allows the editing of ACID metadata, as far as
+ * our interest goes: setting the type to one-shot prevents the setting of BPM.
+ * Conversely, loops must have their BPM set (normally by setting the number of
+ * beats in the loop, but provided that the file duration is available, this is
+ * equivalent).
+ */
 struct AcidizerTags
 {
-   const double bpm;
+   struct OneShot
+   {
+   };
+
+   struct Loop
+   {
+      explicit Loop(double bpm)
+          : bpm { bpm }
+      {
+      }
+      const double bpm;
+   };
+
+   AcidizerTags(OneShot)
+       : isOneShot { true }
+   {
+   }
+
+   AcidizerTags(Loop loop)
+       : bpm { loop.bpm }
+       , isOneShot { false }
+   {
+   }
+
+   const std::optional<double> bpm;
    const bool isOneShot;
```

Comment:
This suggests to me, maybe you might like instead, simply
`using AcidizerTags = std::variant<OneShot, Loop>;`

Reply:
or not...